stack/core/src/main/java/org/apache/usergrid/corepersistence/pipeline/PipelineContext.java

```diff
@@ -43,16 +47,24 @@
     // it can happen if ES was not updated or has yet to be updated.
     private final boolean keepStaleEntries;
     private String query;
+    private ParsedQuery parsedQuery;
 
 
     public PipelineContext( final ApplicationScope applicationScope, final RequestCursor requestCursor, final int limit, final int id, boolean keepStaleEntries, String query ) {
 
         this.applicationScope = applicationScope;
-        this.requestCursor = requestCursor;
-        this.limit = limit;
         this.id = id;
         this.keepStaleEntries = keepStaleEntries;
         this.query = query;
+        this.parsedQuery = ParsedQueryBuilder.build(query);
+        if (parsedQuery != null && parsedQuery.isDirectQuery()) {
```

Comment:
I can tell the direct query is allowing the request to specify a bunch of UUIDs to batch fetch it seems, or something of that nature. We should open a JIRA with the details and link to this PR so that others can be aware of the new feature.

Reply:
https://issues.apache.org/jira/browse/USERGRID-1349

Reply:
Direct query introduces an operator 'direct' so that a ql with multiple UUIDs and names hits C* only and not ES